powdr/src/lib.rs

```diff
@@ -125,7 +125,12 @@ impl Session {
         }
     }
 
-    pub fn write<S: serde::Serialize>(self, channel: u32, data: &S) -> Self {
+    pub fn write<S: serde::Serialize>(&mut self, data: &S) {
+        let bytes = bincode::serialize(&data).unwrap();
+        self.pipeline.add_prover_data(bytes);
+    }
+
+    pub fn write_fd<S: serde::Serialize>(self, channel: u32, data: &S) -> Self {
```

Comment:
Not too happy about these 2 functions being similar but ultimately quite different and based on different mechanisms...

Reply:
we can't remove the old method because the new (free) one only works with continuations?

Reply:
decided in call:

Session will only expose `write` and `write_bytes` which depend on continuations and use the more efficient method via initial prover memory